### PR TITLE
Use commander config-syncer and remove KubeD

### PR DESCRIPTION
**Pull request by AmFlint**
## Description

We want to remove KubeD.

KubeD is used to synchronize secrets from one namespace to other namespaces. In our case, it's used for Houston JWT signing certificates, and potentially SSL secrets for custom ingress controllers and OpenShift Users.

Multiple problems with KubeD:
- Slow performance in large clusters since it listens to all secrets and configmaps
- In terms of security, it's pretty bad since it uses Cluster-level RBAC permissions and can not be scoped to only certain namespaces.

Moreover, KubeD is only used in very specific use cases:
- Openshift users
- Private CA users

Everyone else doesn't need this config synchronization mechanism.

In order to fix this, we added a sync mechanism in Commander to synchronize necessary secrets from one namespace to another, it's more lightweight and in terms of security we can scope these permissions to namespaces inside a namespace pool.

In this Pull Request, I simply replaced "KubeD" with the new commander config-syncer, and made it compatible with namespace pools:
- If namespace pools disabled, creates ClusterRole + ClusterRoleBinding, since the config syncer needs to upload secrets to auto-generated namespaces (and list these namespaces at the cluster-level).
- For better security, with namespace enabled, permissions are scoped to namespaces in the pool and config-syncer will only synchronize secrets from the astronomer release namespace to these namespaces in the pool.

The config syncer can be completely disabled via a value if the user doesn't need it, as explained in the use cases above.

## Related Issues

https://github.com/astronomer/issues/issues/4451

## Testing

I conducted end to end tests in 2 different use cases:
- Namespace Pools disabled
- Namespace Pools enabled

I'll add e2e testing description for QA folks on the issue.

### CODE REVIEW

File: charts/astronomer/values.yaml [hunk 1]
```diff
@@ -238,6 +238,11 @@ houston:
     #   - 2.0.0-alpine3.10-onbuild
     #   channel: stable
 
+configSyncer:
+  enabled: true
+  # default here is to run at midnight every night
+  schedule: "0 0 * * *"
```
Comment (danielhoherd) [inline charts/astronomer/values.yaml:244]: We should avoid the 0 minute of the hour, which is a high-usage minute.

```suggestion
  schedule: "9 0 * * *"
```

Helm doesn't provide a deterministic random function, but [we've worked around it elsewhere by using the adler32sum function to take the namespace as a seed to generate a random-ish value](https://github.com/astronomer/astronomer/blob/dc97103be1/charts/astronomer/templates/houston/houston-configmap.yaml#L282-L287), which we could re-use here if we want to go that far. The general goal is to spread workloads out over time.

File: tests/chart_tests/test_astronomer_config_syncer.py [hunk 2]
```diff
@@ -0,0 +1,197 @@
+from tests.chart_tests.helm_template_generator import render_chart
+import pytest
+from tests import supported_k8s_versions
+
+
+@pytest.mark.parametrize(
+    "kube_version",
+    supported_k8s_versions,
+)
+def test_astronomer_config_syncer_rbac_namespace_pools_disabled(kube_version):
```
Comment (danielhoherd) [inline tests/chart_tests/test_astronomer_config_syncer.py:10]: If you put all these test cases in a class you can parametrize the class, which will DRY up the test cases a bit

Comment (AmFlint) [reply tests/chart_tests/test_astronomer_config_syncer.py]: Done, moved the other test files from this PR too

File: charts/astronomer/values.yaml (hunk 1 above)
Comment (AmFlint) [reply charts/astronomer/values.yaml]: Thanks for this, I added some "custom logic" for this specific request in a _helper Helm template:
- If schedule is provided by user -> use it
- Else, generate a random schedule (minutes and hours) based on the logic you showed me.

File: charts/astronomer/templates/config-syncer/config-syncer-cronjob.yaml [hunk 3]
```diff
@@ -13,7 +13,7 @@ metadata:
     release: {{ .Release.Name }}
     heritage: {{ .Release.Service }}
 spec:
-  schedule: "{{ .Values.configSyncer.schedule }}"
+  schedule: {{ include "configSyncer.schedule" .}}
```
Comment (danielhoherd) [inline charts/astronomer/templates/config-syncer/config-syncer-cronjob.yaml:16]: Can we get a test for this? This is how we did it with the other adler32sum minute. <https://github.com/astronomer/astronomer/blob/c66c8f75e4/tests/chart_tests/test_houston_configmap.py#L124-L175>

Comment (AmFlint) [reply charts/astronomer/templates/config-syncer/config-syncer-cronjob.yaml:16]: Done

File: charts/astronomer/values.yaml [hunk 4]
```diff
@@ -238,6 +238,11 @@ houston:
     #   - 2.0.0-alpine3.10-onbuild
     #   channel: stable
 
+configSyncer:
+  enabled: true
+  # default here is to run at midnight every night
```
Comment (danielhoherd) [inline charts/astronomer/values.yaml:243]:
```suggestion
  # default here is to run at a pseudo-random splayed minute once a day
```

**Review by danielhoherd — APPROVED** [review]
LGTM! So happy to see kubed being retired.

There are still a many references to kubed in the chart, so we'll need to either fix those before merging this, or do a follow-up PR to remove those references:

```
✔  [remove_kubed_use_config_syncer|✔] $ git grep -i kubed
.circleci/config.yml:          astronomer-tags: "platform postgresql monitoring logging kubed"
.circleci/config.yml:          astronomer-tags: "platform postgresql monitoring logging kubed"
.circleci/config.yml:          astronomer-tags: "platform postgresql monitoring logging kubed"
.circleci/config.yml:          astronomer-tags: "platform postgresql monitoring logging kubed"
.circleci/config.yml:          astronomer-tags: "platform postgresql monitoring logging kubed"
.circleci/config.yml.j2:          astronomer-tags: "platform postgresql monitoring logging kubed"
README.md:- kubed: leave on
bin/install-platform:kubed=true
bin/install-platform:  kubed=false
bin/install-platform:        kubed=true
bin/install-platform:  --set tags.kubed="${kubed}" \
charts/astronomer/templates/commander/commander-role.yaml:- apiGroups: ["kubed.appscode.com"]
charts/astronomer/templates/houston/api/houston-jwt-certificate-secret.yaml:    "kubed.appscode.com/sync": {{ (printf "platform-release=%s" .Release.Name) | quote }}
charts/astronomer/templates/houston/houston-configmap.yaml:            {{ if .Values.global.kubedEnabled }}
charts/grafana/dashboards/kube-dns.json:          "expr": "avg by (le) (rate(kubedns_probe_kubedns_latency_ms_bucket{pod=~\"$pod\"}[5m]))",
charts/grafana/dashboards/kube-dns.json:          "expr": "sum(rate(kubedns_dnsmasq_misses{pod=~\"$pod\"}[5m]))",
charts/grafana/dashboards/kube-dns.json:          "expr": "sum(rate(kubedns_dnsmasq_hits{pod=~\"$pod\"}[5m]))",
charts/prometheus/templates/prometheus-alerts-configmap.yaml:        - alert: KubeDeploymentGenerationMismatch
charts/prometheus/templates/prometheus-alerts-configmap.yaml:        - alert: KubeDeploymentReplicasMismatch
charts/prometheus/templates/prometheus-alerts-configmap.yaml:        - alert: KubeDaemonSetRolloutStuck
charts/prometheus/templates/prometheus-alerts-configmap.yaml:        - alert: KubeDaemonSetNotScheduled
charts/prometheus/templates/prometheus-alerts-configmap.yaml:        - alert: KubeDaemonSetMisScheduled
charts/prometheus/templates/prometheus-config-configmap.yaml:            regex: "^sidecar|^kubedns"
configs/dev.yaml:## Kubed configuration
configs/dev.yaml:kubed:
configs/dev.yaml:    kubed:
configs/dev.yaml:      repository: quay.io/astronomer/ap-kubed
configs/local-dev-ha.yaml:  kubed: true
configs/local-dev.yaml:  kubed: true
configs/master.yaml:## Kubed configuration
configs/master.yaml:kubed:
configs/master.yaml:    kubed:
configs/master.yaml:      repository: quay.io/astronomer/ap-kubed
configs/nginx-only.yaml:  kubed: false
configs/single-namespace.yaml:  kubed: false
tests/chart_tests/test_houston_configmap.py:def test_houston_configmap_with_kubed_enabled():
tests/chart_tests/test_houston_configmap.py:    """Validate the houston configmap and its embedded data with kubedEnabled."""
tests/chart_tests/test_houston_configmap.py:        values={"global": {"kubedEnabled": True}},
tests/chart_tests/test_houston_configmap.py:def test_houston_configmap_with_kubed_disabled():
tests/chart_tests/test_houston_configmap.py:    """Validate the houston configmap and its embedded data with kubedEnabled."""
tests/chart_tests/test_houston_configmap.py:        values={"global": {"kubedEnabled": False}},
tests/functional_tests/test_container_no_root.py:    "astronomer-kubed",
values.yaml:  # Kubed
values.yaml:  kubed: true
values.yaml:## KubeD configuration
values.yaml:kubed:
```

**Review by andriisoldatenko — APPROVED** [review]

**Pull request merged**